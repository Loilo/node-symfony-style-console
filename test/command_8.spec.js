const wrapper = require('./wrapper')
const { TableCell } = require('../dist')

test('08 Ensure formatting tables when using multiple headers with TableCell', wrapper(io => {
    const headers = [
        [new TableCell('Main table title', { colspan: 3 })],
        ['ISBN', 'Title', 'Author'],
    ]

    const rows = [
        [
            '978-0521567817',
            'De Monarchia',
            new TableCell("Dante Alighieri\nspans multiple rows", { rowspan: 2 }),
        ],
        ['978-0804169127', 'Divine Comedy'],
    ]

    io.table(headers, rows)
}))
