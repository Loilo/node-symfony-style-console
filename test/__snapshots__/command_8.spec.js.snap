// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`08 Ensure formatting tables when using multiple headers with TableCell 1`] = `
"---------------- --------------- --------------------- 
  Main table title                                      
 ---------------- --------------- --------------------- 
  ISBN             Title           Author               
 ---------------- --------------- --------------------- 
  978-0521567817   De Monarchia    Dante Alighieri      
  978-0804169127   Divine Comedy   spans multiple rows  
 ---------------- --------------- ---------------------"
`;
